x/gov/tick.go

```diff
@@ -0,0 +1,144 @@
+	for ; votesIterator.Valid(); votesIterator.Next() {
+		vote := &Vote{}
+		keeper.cdc.MustUnmarshalBinary(votesIterator.Value(), vote)
+
+		if val, ok := currValidators[addressToString(vote.Voter)]; ok {
+			val.Vote = vote.Option
+			currValidators[addressToString(vote.Voter)] = val
+		} else {
+			for _, delegation := range keeper.sk.GetDelegations(ctx, vote.Voter, math.MaxInt16) { // TODO: Replace with MaxValidators from Stake params
+				val := currValidators[addressToString(delegation.ValidatorAddr)]
+				val.Minus = val.Minus.Add(delegation.Shares)
+				currValidators[addressToString(delegation.ValidatorAddr)] = val
+
+				validatorPower := val.ValidatorInfo.EquivalentBondedShares(pool)
+				delegatorShare := delegation.Shares.Quo(val.ValidatorInfo.DelegatorShares)
+				votingPower := validatorPower.Mul(delegatorShare)
+				results[vote.Option] = results[vote.Option].Add(votingPower)
+				totalVotingPower = totalVotingPower.Add(votingPower)
+			}
+		}
+
+		keeper.deleteVote(ctx, vote.ProposalID, vote.Voter)
+	}
+	votesIterator.Close()
+
+	nonVoting = []sdk.Address{}
+	for _, val := range currValidators {
+		if len(val.Vote) == 0 {
+			nonVoting = append(nonVoting, val.ValidatorInfo.Owner)
+		} else {
+			validatorPower := val.ValidatorInfo.EquivalentBondedShares(pool)
+			sharesAfterMinus := val.ValidatorInfo.DelegatorShares.Sub(val.Minus)
+			percentAfterMinus := sharesAfterMinus.Quo(val.ValidatorInfo.DelegatorShares)
+			votingPower := validatorPower.Mul(percentAfterMinus)
+
+			results[val.Vote] = results[val.Vote].Add(votingPower)
+			totalVotingPower = totalVotingPower.Add(votingPower)
+		}
+	}
+
+	tallyingProcedure := keeper.GetTallyingProcedure(ctx)
+
+	if totalVotingPower.Sub(results["Abstain"]).Equal(sdk.ZeroRat()) {
+		return false, nonVoting
+	} else if results["NoWithVeto"].Quo(totalVotingPower).GT(tallyingProcedure.Veto) {
+		return false, nonVoting
+	} else if results["Yes"].Quo(totalVotingPower.Sub(results["Abstain"])).GT(tallyingProcedure.Threshold) {
+		return true, nonVoting
+	}
+	return false, nonVoting
+}
+
+func shouldPopInactiveProposalQueue(ctx sdk.Context, keeper Keeper) bool {
+	depositProcedure := keeper.GetDepositProcedure(ctx)
+	peekProposal := keeper.InactiveProposalQueuePeek(ctx)
+
+	if peekProposal == nil {
+		return false
+	} else if peekProposal.Status != "DepositPeriod" {
+		return true
+	} else if ctx.BlockHeight() >= peekProposal.SubmitBlock+depositProcedure.MaxDepositPeriod {
+		return true
+	}
+	return false
+}
+
+func shouldPopActiveProposalQueue(ctx sdk.Context, keeper Keeper) bool {
+	votingProcedure := keeper.GetVotingProcedure(ctx)
+	peekProposal := keeper.ActiveProposalQueuePeek(ctx)
+
+	if peekProposal == nil {
+		return false
+	} else if ctx.BlockHeight() >= peekProposal.VotingStartBlock+votingProcedure.VotingPeriod {
+		return true
+	}
+	return false
+}
+
+func addressToString(addr sdk.Address) string {
+	return fmt.Sprintf("%s", addr.Bytes())
+}
```

Comment:
eliminate this function, can be replaced with `string(addr)` everywhere (address is a bytes type do you don't need to call `.Bytes()`

Reply:
:open_mouth: